### ISSUE TITLE
Intel oneapi/2024.2.0: `config/meson.build:78:4: ERROR: Problem encountered: MKL not supported for this compiler`.

### ISSUE THREAD

**Issue by hongyi-zhao** [dftd4/dftd4#252]
On Ubuntu 22.04.4 LTS, I tried to compile dftd4  `main` branch using intel oneapi/2024.2.0 and I noticed that the lapack package shipped with MKL cannot be used, so I must do the compilation as below:

```shell
$ module load oneapi/2024.2.0
$ sudo apt -y install liblapack-dev
$ pyenv shell dftd4
$ pip list
Package              Version
-------------------- -------
beautifulsoup4       4.12.3
cffi                 1.16.0
FORD                 7.0.5
graphviz             0.20.1
iniconfig            2.0.0
Jinja2               3.1.3
Markdown             3.4.4
markdown-include     0.7.2
markdown-it-py       3.0.0
MarkupSafe           2.1.5
mdurl                0.1.2
meson                1.5.1
packaging            24.1
pcpp                 1.30
pip                  24.2
pluggy               1.5.0
pycparser            2.21
Pygments             2.17.2
pytest               8.3.2
python-markdown-math 0.8
rich                 13.7.0
setuptools           65.5.0
soupsieve            2.5
toposort             1.7
tqdm                 4.64.1

# The following steps are run from the top level root directory of the local dftd4 repository  
$ CC=icx CXX=icpx FC=ifx MPICC=mpiicx MPICXX=mpiicpx MPIFC=mpiifx proxychains-ng-socks5 meson setup _build -Dapi_v2=true -Dpython=true -Dpython_version=$(pyenv which python)
Version: 1.5.1
Source dir: /home/werner/Public/repo/github.com/dftd4/dftd4.git
Build dir: /home/werner/Public/repo/github.com/dftd4/dftd4.git/_build
Build type: native build
Project name: dftd4
Project version: 3.6.0
Fortran compiler for the host machine: ifx (intel-llvm 2024.2.0 "ifx (IFX) 2024.2.0 20240602")
Fortran linker for the host machine: ifx ld.bfd 2.38
Host machine cpu family: x86_64
Host machine cpu: x86_64
C compiler for the host machine: icx (intel-llvm 2024.2.0 "Intel(R) oneAPI DPC++/C++ Compiler 2024.2.0 (2024.2.0.20240602)")
C linker for the host machine: icx ld.bfd 2.38
Run-time dependency OpenMP found: YES 5.1
Found pkg-config: YES (/usr/bin/pkg-config) 0.29.2
Run-time dependency lapack found: YES 3.10.0
Library blas found: YES
Found CMake: /home/werner/.pyenv/shims/cmake (3.22.1)
Run-time dependency mctc-lib found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency mctc-lib

Executing subproject mctc-lib 

mctc-lib| Project name: mctc-lib
mctc-lib| Project version: 0.3.0
mctc-lib| Fortran compiler for the host machine: ifx (intel-llvm 2024.2.0 "ifx (IFX) 2024.2.0 20240602")
mctc-lib| Fortran linker for the host machine: ifx ld.bfd 2.38
mctc-lib| Dependency openmp found: YES 5.1 (cached)
mctc-lib| Run-time dependency json-fortran found: NO (tried pkgconfig and cmake)
mctc-lib| Looking for a fallback subproject for the dependency json-fortran
mctc-lib| Building fallback subproject with default_library=static
mctc-lib| Using subprojects/mctc-lib/subprojects/json-fortran-8.2.5.wrap
mctc-lib| Downloading json-fortran-8.2.5 source from https://github.com/jacobwilliams/json-fortran/archive/refs/tags/8.2.5.tar.gz
Downloading file of unknown size.

Executing subproject mctc-lib:json-fortran-8.2.5

json-fortran-8.2.5| Project name: jsonfortran
json-fortran-8.2.5| Project version: 8.2.5
json-fortran-8.2.5| Fortran compiler for the host machine: ifx (intel-llvm 2024.2.0 "ifx (IFX) 2024.2.0 20240602")
json-fortran-8.2.5| Fortran linker for the host machine: ifx ld.bfd 2.38
json-fortran-8.2.5| Build targets in project: 1
json-fortran-8.2.5| Subproject json-fortran-8.2.5 finished.

mctc-lib| Dependency json-fortran from subproject subprojects/json-fortran-8.2.5 found: YES 8.2.5
mctc-lib| Build targets in project: 4
mctc-lib| Subproject mctc-lib finished.

Dependency mctc-lib from subproject subprojects/mctc-lib found: YES 0.3.0
Run-time dependency multicharge found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency multicharge

Executing subproject multicharge 

multicharge| Project name: multicharge
multicharge| Project version: 0.2.0
multicharge| Fortran compiler for the host machine: ifx (intel-llvm 2024.2.0 "ifx (IFX) 2024.2.0 20240602")
multicharge| Fortran linker for the host machine: ifx ld.bfd 2.38
multicharge| Dependency openmp found: YES 5.1 (cached)
multicharge| Dependency lapack found: YES 3.10.0 (cached)
multicharge| Run-time dependency blas found: YES 3.10.0
multicharge| Dependency mctc-lib found: YES 0.3.0 (cached)
multicharge| Run-time dependency mstore found: NO (tried pkgconfig and cmake)
multicharge| Looking for a fallback subproject for the dependency mstore

Executing subproject multicharge:mstore

mstore| Project name: mstore
mstore| Project version: 0.2.0
mstore| Fortran compiler for the host machine: ifx (intel-llvm 2024.2.0 "ifx (IFX) 2024.2.0 20240602")
mstore| Fortran linker for the host machine: ifx ld.bfd 2.38
mstore| Dependency openmp found: YES 5.1 (cached)
mstore| Dependency mctc-lib found: YES 0.3.0 (cached)
mstore| Build targets in project: 9
mstore| Subproject mstore finished.

multicharge| Dependency mstore from subproject subprojects/mstore found: YES 0.2.0
multicharge| Build targets in project: 10
multicharge| Subproject multicharge finished.

Dependency multicharge from subproject subprojects/multicharge found: YES 0.2.0
Program tester.py found: YES (/home/werner/Public/repo/github.com/dftd4/dftd4.git/app/tester.py)
Program config/install-mod.py found: YES (/home/werner/Public/repo/github.com/dftd4/dftd4.git/config/install-mod.py)
Program asciidoctor found: YES (/usr/local/bin/asciidoctor)
Configuring dftd4.1 with command
Dependency mstore found: YES 0.2.0 (cached)
Program /home/werner/.pyenv/versions/dftd4/bin/python (cffi) found: YES (/home/werner/.pyenv/versions/dftd4/bin/python) modules: cffi
Run-time dependency python found: YES 3.11
Configuring _libdftd4.h with command
Configuring _libdftd4.c with command
Build targets in project: 17

dftd4 3.6.0

  Subprojects
    json-fortran-8.2.5: YES (from mctc-lib)
    mctc-lib          : YES
    mstore            : YES (from multicharge)
    multicharge       : YES

  User defined options
    api_v2            : true
    python            : true
    python_version    : /home/werner/.pyenv/versions/dftd4/bin/python

Found ninja-1.10.1 at /usr/bin/ninja

$ meson test -C _build --print-errorlogs
ninja: Entering directory `/home/werner/Public/repo/github.com/dftd4/dftd4.git/_build'
[1/159] Compiling C object test/api/api-tester.p/example.c.o
icx: warning: Use of '-qopenmp' recommended over '-fopenmp' [-Wrecommended-option]
[113/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_data_en.f90.o
../src/dftd4/data/en.f90(68): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_electronegativity_sym
--------------------^
../src/dftd4/data/en.f90(83): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_electronegativity_num
--------------------^
[117/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_data_covrad.f90.o
../src/dftd4/data/covrad.f90(72): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_covalent_rad_sym
--------------------^
../src/dftd4/data/covrad.f90(87): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_covalent_rad_num
--------------------^
[119/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_version.f90.o
../src/dftd4/version.f90(38): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_dftd4_version
--------------------^
[121/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_charge.f90.o
../src/dftd4/charge.f90(35): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_charges
--------------------^
[123/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_utils.f90.o
../src/dftd4/utils.f90(29): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: wrap_to_central_cell
--------------------^
[124/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_cutoff.f90.o
../src/dftd4/cutoff.f90(121): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_lattice_points_cutoff
--------------------^
[126/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_ncoord.f90.o
../src/dftd4/ncoord.f90(45): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_coordination_number
--------------------^
[130/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_damping_rational.f90.o
../src/dftd4/damping/rational.f90(61): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_dispersion2
--------------------^
../src/dftd4/damping/rational.f90(282): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_dispersion3
--------------------^
../src/dftd4/damping/rational.f90(332): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_pairwise_dispersion2
--------------------^
../src/dftd4/damping/rational.f90(397): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_pairwise_dispersion3
--------------------^
[133/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_model.f90.o
../src/dftd4/model.f90(115): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: d4_ref
--------------------^
../src/dftd4/model.f90(128): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: new_d4_model_with_checks
--------------------^
../src/dftd4/model.f90(278): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: new_d4_model_no_checks
--------------------^
../src/dftd4/model.f90(417): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: weight_references
--------------------^
../src/dftd4/model.f90(548): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_atomic_c6
--------------------^
../src/dftd4/model.f90(645): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_polarizibilities
--------------------^
[136/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_disp.f90.o
../src/dftd4/disp.f90(40): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_dispersion
--------------------^
../src/dftd4/disp.f90(129): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_properties
--------------------^
../src/dftd4/disp.f90(173): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_pairwise_dispersion
--------------------^
[137/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_numdiff.f90.o
../src/dftd4/numdiff.f90(36): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_dispersion_hessian
--------------------^
[138/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_param.f90.o
../src/dftd4/param.f90(99): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_functionals
--------------------^
../src/dftd4/param.f90(244): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_rational_damping_name
--------------------^
../src/dftd4/param.f90(265): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_rational_damping_id
--------------------^
[139/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_output.f90.o
../src/dftd4/output.f90(39): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: ascii_atomic_radii
--------------------^
../src/dftd4/output.f90(72): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: ascii_atomic_references
--------------------^
../src/dftd4/output.f90(121): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: ascii_system_properties
--------------------^
../src/dftd4/output.f90(173): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: ascii_results
--------------------^
../src/dftd4/output.f90(225): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: ascii_pairwise
--------------------^
../src/dftd4/output.f90(277): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: ascii_damping_param
--------------------^
../src/dftd4/output.f90(312): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: turbomole_gradlatt
--------------------^
../src/dftd4/output.f90(402): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: turbomole_gradient
--------------------^
../src/dftd4/output.f90(514): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: json_results
--------------------^
../src/dftd4/output.f90(630): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: tagged_result
--------------------^
[142/159] Compiling Fortran object libdftd4.so.3.6.0.p/src_dftd4_api.f90.o
../src/dftd4/api.f90(92): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_version_api
--------------------^
../src/dftd4/api.f90(106): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: new_error_api
--------------------^
../src/dftd4/api.f90(121): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: delete_error_api
--------------------^
../src/dftd4/api.f90(140): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: check_error_api
--------------------^
../src/dftd4/api.f90(165): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_error_api
--------------------^
../src/dftd4/api.f90(195): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: new_structure_api
--------------------^
../src/dftd4/api.f90(244): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: delete_structure_api
--------------------^
../src/dftd4/api.f90(263): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: update_structure_api
--------------------^
../src/dftd4/api.f90(306): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: new_d4_model_api
--------------------^
../src/dftd4/api.f90(342): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: custom_d4_model_api
--------------------^
../src/dftd4/api.f90(380): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: delete_model_api
--------------------^
../src/dftd4/api.f90(400): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: new_rational_damping_api
--------------------^
../src/dftd4/api.f90(434): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: load_rational_damping_api
--------------------^
../src/dftd4/api.f90(471): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: delete_param_api
--------------------^
../src/dftd4/api.f90(491): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_dispersion_api
--------------------^
../src/dftd4/api.f90(571): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_numerical_hessian_api
--------------------^
../src/dftd4/api.f90(627): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_pairwise_dispersion_api
--------------------^
../src/dftd4/api.f90(682): remark #7841: DLL IMPORT/EXPORT is not supported on this platform.   [DLLEXPORT]
   !DEC$ ATTRIBUTES DLLEXPORT :: get_properties_api
--------------------^
[149/159] Linking target test/api/api-tester
icx: warning: Use of '-qopenmp' recommended over '-fopenmp' [-Wrecommended-option]
icx: warning: Use of '-qopenmp' recommended over '-fopenmp' [-Wrecommended-option]
icx: warning: Use of '-qopenmp' recommended over '-fopenmp' [-Wrecommended-option]
[159/159] Linking target test/unit/tester
 1/49 dftd4 / app-version                OK              0.23s
 2/49 dftd4 / app-help                   OK              0.23s
 3/49 dftd4 / app-help-run               OK              0.22s
 4/49 dftd4 / app-help-param             OK              0.21s
 5/49 dftd4 / app-param-list             OK              0.20s
 6/49 dftd4 / app-license                OK              0.19s
 7/49 dftd4 / app-citation               OK              0.18s
 8/49 mctc-lib / math                    OK              0.45s
 9/49 mctc-lib / read                    OK              0.45s
10/49 mctc-lib / read-aims               OK              0.44s
11/49 mctc-lib / read-ctfile             OK              0.43s
12/49 mctc-lib / read-gaussian           OK              0.43s
13/49 mctc-lib / read-genformat          OK              0.42s
14/49 mctc-lib / read-pdb                OK              0.41s
15/49 mctc-lib / read-qchem              OK              0.40s
16/49 mctc-lib / read-qcschema           OK              0.40s
17/49 mctc-lib / read-turbomole          OK              0.39s
18/49 mctc-lib / read-vasp               OK              0.38s
19/49 mctc-lib / read-xyz                OK              0.38s
20/49 mctc-lib / symbols                 OK              0.37s
21/49 mctc-lib / write                   OK              0.37s
22/49 mctc-lib / write-aims              OK              0.36s
23/49 mctc-lib / write-ctfile            OK              0.35s
24/49 mctc-lib / write-gaussian          OK              0.35s
25/49 mctc-lib / write-genformat         OK              0.34s
26/49 mctc-lib / write-pdb               OK              0.34s
27/49 mctc-lib / write-qchem             OK              0.33s
28/49 mctc-lib / write-turbomole         OK              0.32s
29/49 mctc-lib / write-vasp              OK              0.31s
30/49 mctc-lib / write-xyz               OK              0.31s
31/49 multicharge / wignerseitz          OK              0.28s
32/49 dftd4 / app-noargs                 EXPECTEDFAIL    0.20s   exit status 128
33/49 dftd4 / api-test                   OK              0.19s
34/49 multicharge / ncoord               OK              0.45s
35/49 dftd4 / pairwise                   OK              0.35s
36/49 dftd4 / app-r2scan3c-mod           OK              0.42s
37/49 dftd4 / app-r2scan3c-cli           OK              0.42s
38/49 dftd4 / app-gradient               OK              0.51s
39/49 dftd4 / model                      OK              0.45s
40/49 dftd4 / app-pair-analysis          OK              0.51s
41/49 dftd4 / app-r2scan3c               OK              0.55s
42/49 dftd4 / app-properties             OK              0.59s
43/49 dftd4 / ncoord                     OK              0.95s
44/49 multicharge / pbc                  OK              1.35s
45/49 multicharge / model                OK              1.96s
46/49 dftd4 / app-energy                 OK              1.91s
47/49 dftd4 / param                      OK              1.80s
48/49 dftd4 / periodic                   OK              1.98s
49/49 dftd4 / dftd4                      FAIL            2.62s   exit status 1
>>> MSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 LD_LIBRARY_PATH=/home/werner/Public/repo/github.com/dftd4/dftd4.git/_build/:/opt/intel/oneapi/2024.2.0/tbb/2021.13/env/../lib/intel64/gcc4.8:/opt/intel/oneapi/2024.2.0/mpi/2021.13/opt/mpi/libfabric/lib:/opt/intel/oneapi/2024.2.0/mpi/2021.13/lib:/opt/intel/oneapi/2024.2.0/mkl/2024.2/lib:/opt/intel/oneapi/2024.2.0/ippcp/2021.12/lib:/opt/intel/oneapi/2024.2.0/ipp/2021.12/lib:/opt/intel/oneapi/2024.2.0/dpl/2022.6/lib:/opt/intel/oneapi/2024.2.0/dnnl/2024.2/lib:/opt/intel/oneapi/2024.2.0/debugger/2024.2/opt/debugger/lib:/opt/intel/oneapi/2024.2.0/dal/2024.5/lib:/opt/intel/oneapi/2024.2.0/compiler/2024.2/opt/oclfpga/host/linux64/lib:/opt/intel/oneapi/2024.2.0/compiler/2024.2/opt/compiler/lib:/opt/intel/oneapi/2024.2.0/compiler/2024.2/lib MESON_TEST_ITERATION=1 MALLOC_PERTURB_=10 ASAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1 /home/werner/Public/repo/github.com/dftd4/dftd4.git/_build/test/unit/tester dftd4
―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
stdout:
 -2.438100727380593E-003
 -0.181515174569810     
stderr:
# Testing: dftd4
  Starting PBE-D4 ... (1/19)
       ... PBE-D4 [PASSED]
  Starting B97-D4 ... (2/19)
       ... B97-D4 [PASSED]
  Starting TPSS-D4 ... (3/19)
       ... TPSS-D4 [PASSED]
  Starting PWPB95-D4 ... (4/19)
       ... PWPB95-D4 [PASSED]
  Starting B2PLYP-D4 ... (5/19)
       ... B2PLYP-D4 [PASSED]
  Starting PW6B95-D4 ... (6/19)
       ... PW6B95-D4 [PASSED]
  Starting OLYP-D4 ... (7/19)
       ... OLYP-D4 [PASSED]
  Starting PBE0-D4 ... (8/19)
       ... PBE0-D4 [PASSED]
  Starting RPBE-D4-ATM ... (9/19)
       ... RPBE-D4-ATM [PASSED]
  Starting B2GPPLYP-D4-ATM ... (10/19)
       ... B2GPPLYP-D4-ATM [PASSED]
  Starting LH14t-calPBE-D4-ATM ... (11/19)
       ... LH14t-calPBE-D4-ATM [PASSED]
  Starting B1B95-D4-ATM ... (12/19)
       ... B1B95-D4-ATM [PASSED]
  Starting M06L-D4-ATM ... (13/19)
       ... M06L-D4-ATM [PASSED]
  Starting TPSSh-D4-ATM ... (14/19)
       ... TPSSh-D4-ATM [PASSED]
  Starting TPSSh-D4-ATM-AmF3 ... (15/19)
       ... TPSSh-D4-ATM-AmF3 [FAILED]
  Message: Floating point value missmatch
  Starting HF-D4-ATM ... (16/19)
       ... HF-D4-ATM [PASSED]
  Starting CAM-B3LYP-D4-ATM ... (17/19)
       ... CAM-B3LYP-D4-ATM [PASSED]
  Starting r2SCAN-3c ... (18/19)
       ... r2SCAN-3c [PASSED]
  Starting Actinides ... (19/19)
       ... Actinides [FAILED]
  Message: Floating point value missmatch
2 test(s) failed!
1
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――


Summary of Failures:

49/49 dftd4 / dftd4              FAIL            2.62s   exit status 1

Ok:                 47  
Expected Fail:      1   
Fail:               1   
Unexpected Pass:    0   
Skipped:            0   
Timeout:            0   

Full log written to /home/werner/Public/repo/github.com/dftd4/dftd4.git/_build/meson-logs/testlog.txt
```

So, I wonder how to solve the following problems:

1. How do we let dftd4 pick the lapack package shipped with MKL?
2. There is a failed test, and I would like to know how to fix it.

Any suggestions or comments you might have for fixing these problems will be helpful.

Regards,
Zhao

**Comment by awvwgk** [comment]
Lukas @lukaswittmann, Marvin @marvinfriede, can you have a look and check why the Actinide support here is broken?

**Comment by marvinfriede** [comment]
@hongyi-zhao Are you using the latest version of `multicharge`? There was an update to the actinide parameters in `multicharge`, which changed the reference values in DFT-D4.

**Comment by hongyi-zhao** [comment]
@marvinfriede 
1. In the above report, the `multicharge` package is installed by meson automatically and it's the latest development version as shown below:

```shell
werner@x13dai-t:~/Public/repo/github.com/dftd4/dftd4.git/subprojects$ cat multicharge.wrap 
[wrap-git]
directory = multicharge
url = https://github.com/grimme-lab/multicharge
revision = head
```

2. I still wonder how I can use the MKL to provide all mathematical libraries, say, `lapack`, used by dftd4.

**Comment by marvinfriede** [comment]
Is it really the latest release or the latest commit? I believe it should fail with the latest release, but work with the latest commit. The latest release does not have the changed parameters, which are used for the D4 regression tests.
Meson should get the latest commit (HEAD) if it clones from GitHub during the setup.

**Comment by hongyi-zhao** [comment]
You are right. It's the latest commit:

```shell
(dftd4) werner@x13dai-t:~/Public/repo/github.com/dftd4/dftd4.git/subprojects/multicharge$ git status 
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   config/meson.build

no changes added to commit (use "git add" and/or "git commit -a")
(dftd4) werner@x13dai-t:~/Public/repo/github.com/dftd4/dftd4.git/subprojects/multicharge$ git log -1| head
commit 92b37fbdb85a9a310b137d3f18410f66d37fb7a5
Author: Marcel Mueller <marcel.mueller@thch.uni-bonn.de>
Date:   Wed Jul 31 23:13:04 2024 +0200

    Fix continuous integration (#31)
    
    * update GH workflow to more recent Python version and Intel env vars
    
    Signed-off-by: Marcel Müller <marcel.mueller@thch.uni-bonn.de>
```

**Comment by hongyi-zhao** [comment]
I've created a PR to fix this problem: https://github.com/dftd4/dftd4/pull/253.
